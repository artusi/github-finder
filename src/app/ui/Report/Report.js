import React from "react";
import PropTypes from "prop-types";
import styles from "./Report.module.css";

export function Report({ id, customClassName, list }) {
  if (!list.length) return null;

  return (
    <table
      id={`report-${id}`}
      className={`${styles.report} ${customClassName}`}
    >
      <tbody>
        {list.map(item => {
          const rowId = item.id;
          return (
            <tr key={rowId} className={`${styles.row} ${item.customClass}`}>
              {item.cells.map((cell, index) => {
                const cellKey = rowId + index;
                return (
                  <td
                    className={`${styles.cell} ${cell.customClass}`}
                    key={cellKey}
                  >
                    {cell.content}
                  </td>
                );
              })}
            </tr>
          );
        })}
      </tbody>
    </table>
  );
}

Report.propTypes = {
  id: PropTypes.string.isRequired,
  list: PropTypes.array.isRequired,
  customClassName: PropTypes.string
};

Report.defaultProps = {
  customClassName: ""
};

export default Report;
