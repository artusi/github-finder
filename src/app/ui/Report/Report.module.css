.report {
  margin: 0 auto;
  width: 100%;
  font-size: 12px;
  color: #00324d;
}
.report a {
  color: #00324d;
}
.report a:hover {
  text-decoration: none;
}
.row:nth-child(odd) {
  background: #fff;
}
.row:nth-child(even) {
  background: #ffffffba;
}
.row:hover {
  background: #f3c1bc;
}
.cell {
  padding: 6px;
  box-shadow: 0 1px 3px 0 rgba(51, 51, 51, 0.2);
}
