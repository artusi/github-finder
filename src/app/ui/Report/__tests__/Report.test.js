import React from "react";
import { Report } from "../Report";

const initialProps = {
  id: "id",
  list: []
};

describe("<Report /> - render", () => {
  it("renders", () => {
    const wrapper = shallow(<Report {...initialProps} />);

    expect(wrapper).toMatchSnapshot();
  });

  it("renders with all props > prefix | defaultValue", () => {
    const allProps = {
      id: "id",
      list: [
        {
          id: "row1",
          cells: [
            { content: "1 content", customClass: "1 class" },
            { content: "2 content", customClass: "2class" }
          ]
        }
      ]
    };
    const wrapper = shallow(<Report {...allProps} />);

    expect(wrapper).toMatchSnapshot();
  });
});
