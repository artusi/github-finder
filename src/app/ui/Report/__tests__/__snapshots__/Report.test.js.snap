// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Report /> - render renders 1`] = `""`;

exports[`<Report /> - render renders with all props > prefix | defaultValue 1`] = `
<table
  className="report "
  id="report-id"
>
  <tbody>
    <tr
      className="row undefined"
      key="row1"
    >
      <td
        className="cell 1 class"
        key="row10"
      >
        1 content
      </td>
      <td
        className="cell 2class"
        key="row11"
      >
        2 content
      </td>
    </tr>
  </tbody>
</table>
`;
